app/Http/Controllers/GazetteerController.php

```diff
@@ -228,7 +228,9 @@ private function searchDataitems($parameters)
             ->with(['datasource' => function ($q) {
                 $q->select('id', 'name', 'description', 'link');
             }])
-            ->whereIn('datasource_id', $datasourceIDs);
+            ->whereIn('datasource_id', $datasourceIDs)
+            ->join('tlcmap.recordtype', 'tlcmap.dataitem.recordtype_id', '=', 'tlcmap.recordtype.id')
+            ->select('tlcmap.dataitem.*', 'tlcmap.recordtype.type as recordtype_type');

```

Comment:
This solutions seem a little bit clumsy. See the next comment for suggested solutions.